projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/serverRequests.ts

```diff
@@ -47,6 +63,66 @@ type jobRequestResult = {
   isSuccessful: boolean;
 };
 
+type TaskStatusResult = {
+  /**
+   * Type of the task
+   */
+  task_type: string;
+  /**
+   * Status of the task
+   */
+  status: string;
+  /**
+   * Result message of the task if no errors occurred
+   */
+  message: string | IJobConvertToNotebookMessage;
+  /**
+   * Error message
+   */
+  error: string;
+};
+
+const pollForTaskCompletion = async (
+  taskId: string,
+  maxAttempts = 43200,
+  interval = 10000,
+  errorThreshold = 5
+): Promise<TaskStatusResult> => {
+  let errorCounter = 0;
```

Comment:
Why do we count errors here? 
What type of errors are expected?

Reply:
My intuition here is that we do polling to get the status of the job and if there is some temporary (connectivity for example) issue while a task is in progress, then we would give it a retry for some time until we eventually stop trying (5 attempts). That's why I introduced this error threshold for such cases.

Reply:
@yonitoo I understand, but i think we somehow should manage those errors, what if it is some other essential error in this way we might lose track of it?

Reply:
Well, then we can initially just make it fail instantly and later change it.

Reply:
Made it this way.